packages/core-transaction-pool/src/memory.ts

```diff
@@ -76,70 +96,82 @@ export class Memory {
         }
 
         return new Set();
     }
 
-    public remember(MemoryTransaction: MemoryTransaction, maxTransactionAge: number, databaseReady?: boolean): void {
-        const transaction: Interfaces.ITransaction = MemoryTransaction.transaction;
+    public remember(SequentialTransaction: SequentialTransaction, maxTransactionAge: number, databaseReady?: boolean): void {
+        const transaction: Interfaces.ITransaction = SequentialTransaction.transaction;
 
         assert.strictEqual(this.byId[transaction.id], undefined);
 
         if (databaseReady) {
             // Sequence is provided from outside, make sure we avoid duplicates
             // later when we start using our this.sequence.
-            assert.strictEqual(typeof MemoryTransaction.sequence, "number");
+            assert.strictEqual(typeof SequentialTransaction.sequence, "number");
 
-            this.sequence = Math.max(this.sequence, MemoryTransaction.sequence) + 1;
+            this.sequence = Math.max(this.sequence, SequentialTransaction.sequence) + 1;
         } else {
             // Sequence should only be set during DB load (when sequences come
             // from the database). In other scenarios sequence is not set and we
             // set it here.
-            MemoryTransaction.sequence = this.sequence++;
+            SequentialTransaction.sequence = this.sequence++;
         }
 
-        this.all.push(MemoryTransaction);
+        this.all.push(SequentialTransaction);
         this.allIsSorted = false;
 
-        this.byId[transaction.id] = MemoryTransaction;
+        this.byId[transaction.id] = SequentialTransaction;
 
         const sender: string = transaction.data.senderPublicKey;
         const type: number = transaction.type;
 
         if (this.bySender[sender] === undefined) {
             // First transaction from this sender, create a new Set.
-            this.bySender[sender] = new Set([MemoryTransaction]);
+            this.bySender[sender] = new Set([SequentialTransaction]);
         } else {
             // Append to existing transaction ids for this sender.
-            this.bySender[sender].add(MemoryTransaction);
+            this.bySender[sender].add(SequentialTransaction);
         }
 
         if (this.byType[type] === undefined) {
             // First transaction of this type, create a new Set.
-            this.byType[type] = new Set([MemoryTransaction]);
+            this.byType[type] = new Set([SequentialTransaction]);
         } else {
             // Append to existing transaction ids for this type.
-            this.byType[type].add(MemoryTransaction);
+            this.byType[type].add(SequentialTransaction);
         }
 
-        if (MemoryTransaction.expiresAt(maxTransactionAge) !== null) {
-            this.byExpiration.push(MemoryTransaction);
+        if (type !== Enums.TransactionTypes.TimelockTransfer) {
+            const currentHeight: number = this.currentHeight();
+            if (currentHeight) {
```

Comment:
This check can be removed, there's now always a block and therefore height always truthy.

Reply:
Y